server/user.go

```diff
@@ -299,36 +318,48 @@ func (s *UserMgmtServer) getCreds(r *http.Request, requiresAdmin bool) (api.Cred
 		return api.Creds{}, api.ErrorUnauthorized
 	}
 
-	usr, err := s.um.Get(sub)
-	if err != nil {
-		if err == user.ErrorNotFound {
-			return api.Creds{}, api.ErrorUnauthorized
-		}
-		log.Errorf("userMgmtServer: GetCreds err: %q", err)
-		return api.Creds{}, err
-	}
-
-	i := 0
-	for _, clientID := range clientIDs {
-		// Make sure the client actually exists.
-		isAdmin, err := s.cm.IsDexAdmin(clientID)
+	var usr user.User
+	if s.allowClientCredsAuth && (len(clientIDs) == 1) && (sub == clientIDs[0]) {
```

Comment:
yep, it's a bearer token obtained from Dex /token endpoint, using grant_type=client_credentials and auth basic (urlencode_base64 client id+secret).
then we extract the clientID from aud and verify it's an admin.

what kind of auth do you want then?